### PR TITLE
Feat/imgix base url implementation

### PR DESCRIPTION
**Pull request by syedbadshah-toptal**
base URL implementation with unit test cases.
system objects meta data compiled in a correct way. Added **imgixBaseURL** custom preference as well.

### CODE REVIEW

**Review by frederickfogerty — APPROVED** [review]
LGTM

**Review by frederickfogerty — APPROVED** [review]
LGTM

File: metadata/imgix_data/meta/system-objecttype-extensions.xml [hunk 1]
```diff
@@ -17,13 +17,22 @@
                 <mandatory-flag>false</mandatory-flag>
                 <externally-managed-flag>false</externally-managed-flag>
                 <min-length>0</min-length>
+            </attribute-definition>
+            <attribute-definition attribute-id="imgixBaseURL">
+                <display-name xml:lang="x-default">Imgix Base URL</display-name>
```
Comment (luqven) [inline metadata/imgix_data/meta/system-objecttype-extensions.xml:22]: Nitpick: imgix should never be capitalized
```suggestion
                <display-name xml:lang="x-default">imgix Base URL</display-name>
```

**Review by luqven — DISMISSED** [review]
lgtm, one small non-blocking suggestion

File: metadata/imgix_data/meta/system-objecttype-extensions.xml (hunk 1 above)
Comment (syedbadshah-toptal) [reply metadata/imgix_data/meta/system-objecttype-extensions.xml]: @luqven I've updated the display name as suggestion, thanks.

**Comment by frederickfogerty** [comment]
I'll merge this into `main` tomorrow

**Review by frederickfogerty — APPROVED** [review]
LGTM

**Pull request merged**